### PR TITLE
forge reminder core

### PR DESCRIPTION
**Pull request by Ascynx**
# Additions

<details>
<summary>Events</summary> 

- added a cancel check on events
- added joinWorldEvent
- added EndTickEvent

</details>

<details>
<summary>Features</summary>

- added forge reminder (currently only updates when joining a dwarven mines lobby)

</details>

<details>
<summary>TODO</summary>

- Check Forge Gui content for slot 5 and update of slots (outside of tablist update)

</details>

### CODE REVIEW

File: src/main/java/mod/gate/Gate.java [hunk 1]
```diff
@@ -14,5 +18,10 @@ public void onInitializeClient() {
         LOGGER.info(Reference.NAME+" has been initialized!");
 
         new FeatureRegistry();
+        ConfigManager.init();
```
Comment (DalwynWasTaken) [inline src/main/java/mod/gate/Gate.java:20]: This is on me but we want this above FeatureRegistry so we have actual config to load from

File: src/main/java/mod/gate/Gate.java [hunk 2]
```diff
@@ -14,5 +18,10 @@ public void onInitializeClient() {
...
+
+        ClientTickEvents.END_WORLD_TICK.register((world) -> {
```
Comment (DalwynWasTaken) [inline src/main/java/mod/gate/Gate.java:23]: I dislike this being in the Gate.java file, and I really dislike the v1 in the import name so maybe we inject to where we want to be? Or have this somewhere not here like in a static method of the TickEvent which makes instances of itself

Comment (DalwynWasTaken) [reply src/main/java/mod/gate/Gate.java:23]: I do intend to write this myself, so no need to worry about this

File: src/main/java/mod/gate/core/events/JoinWorldEvent.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package mod.gate.core.events;
+
+import net.minecraft.network.packet.s2c.play.GameJoinS2CPacket;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+public class JoinWorldEvent {
+    public CallbackInfo info;
+
+    //region packet
+    private final GameJoinS2CPacket packet;//might not be useful but including it for now
```
Comment (DalwynWasTaken) [inline src/main/java/mod/gate/core/events/JoinWorldEvent.java]: I dislike your comments being attached, move them up a line like normal comments

File: src/main/java/mod/gate/utils/Tablist.java [hunk 4]
```diff
@@ -49,10 +50,20 @@ public static String getArea() {
...
+
+    public static ArrayList<String> getLines() {
+        loadTablist();
+        return Lines;
+    }
+    //endregion
+
     private static void loadTablist() {
-        if (!Scoreboard.isOnSkyblock()) return;
+        if (!Scoreboard.isOnSkyblock()) return;//there's a small issue here
```
Comment (DalwynWasTaken) [inline src/main/java/mod/gate/utils/Tablist.java]: What's the issue? And weird comments again

File: src/main/java/mod/gate/core/mixin/GameJoinMixin.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import mod.gate.core.events.EventHandler;
+import mod.gate.core.events.JoinWorldEvent;
+import net.minecraft.client.network.ClientPlayNetworkHandler;
+import net.minecraft.network.packet.s2c.play.GameJoinS2CPacket;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+@Mixin(ClientPlayNetworkHandler.class)
+public class GameJoinMixin {
```
Comment (DalwynWasTaken) [inline src/main/java/mod/gate/core/mixin/GameJoinMixin.java]: The event is called JoinWorldEvent, What are we going with? MC terminology or a standard we can agree on

**Review by DalwynWasTaken — DISMISSED** [review]

File: src/main/java/mod/gate/utils/Tablist.java (hunk 4 above)
Comment (Ascynx) [reply src/main/java/mod/gate/utils/Tablist.java]: was an issue when I started out as it loaded the tablist before it was accessible

File: src/main/java/mod/gate/Gate.java (hunk 2 above)
Comment (Ascynx) [reply src/main/java/mod/gate/Gate.java:23]: I don't know where to fix that so I'll let that to you

**Review by DalwynWasTaken — APPROVED** [review]
I think everything is done, I shall try my best to get my PR done tonight and tomorrow so this all works

**Pull request merged**